`TestMempool2Nodes` is flaky

https://github.com/rollkit/rollkit/actions/runs/8003210635/job/21858039560?pr=1557#step:4:2702
```
panic: 

mock: Unexpected Method Call
-----------------------------

CheckTx(context.todoCtx,*types.RequestCheckTx)
		0: context.todoCtx{emptyCtx:context.emptyCtx{}}
		1: &types.RequestCheckTx{Tx:[]uint8{0x67, 0x6f, 0x6f, 0x64}, Type:1}

The closest call I have is: 

CheckTx(string,*types.RequestCheckTx)
		0: "mock.Anything"
		1: &types.RequestCheckTx{Tx:[]uint8{0x62, 0x61, 0x64}, Type:0}

Difference found in argument 1:

--- Expected
+++ Actual
@@ -1,2 +1,2 @@
-(*types.RequestCheckTx)(tx:"bad" )
+(*types.RequestCheckTx)(tx:"good" type:RECHECK )
 

Diff: 0: PASS:  (context.todoCtx=context.TODO) == (string=mock.Anything)
	1: FAIL:  (*types.RequestCheckTx=tx:"good" type:RECHECK ) != (*types.RequestCheckTx=tx:"bad" )

goroutine 3960 [running]:
github.com/stretchr/testify/mock.(*Mock).fail(0xc0001758b0, {0x24fca19, 0x6f}, {0xc04d8aad40, 0x4, 0x4})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.4/mock/mock.go:332 +0x1d9
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0001758b0, {0x2c4432d, 0x7}, {0xc042410e60?, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.4/mock/mock.go:492 +0x6bf
github.com/stretchr/testify/mock.(*Mock).Called(0x2?, {0xc042410e60, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.4/mock/mock.go:464 +0x193
github.com/rollkit/rollkit/test/mocks.(*Application).CheckTx(0xc0001758b0, {0x28e87c8?, 0x38abe40}, 0xc042410e40)
	/home/runner/work/rollkit/rollkit/test/mocks/Application.go:49 +0x133
github.com/cometbft/cometbft/abci/client.(*localClient).CheckTxAsync(0xc01bd7f320, {0x28e87c8, 0x38abe40}, 0xc042410e40)
	/home/runner/go/pkg/mod/github.com/cometbft/cometbft@v0.38.5/abci/client/local_client.go:51 +0x156
github.com/cometbft/cometbft/proxy.(*appConnMempool).CheckTxAsync(0xc01bcf7f68, {0x28e87c8, 0x38abe40}, 0x0?)
	/home/runner/go/pkg/mod/github.com/cometbft/cometbft@v0.38.5/proxy/app_conn.go:147 +0x2d6
github.com/rollkit/rollkit/mempool.(*CListMempool).recheckTxs(0xc02172c0d0)
	/home/runner/work/rollkit/rollkit/mempool/clist_mempool.go:656 +0x2c4
github.com/rollkit/rollkit/mempool.(*CListMempool).Update(0xc02172c0d0, 0x2, {0x38abe40, 0x0, 0xc01bd7fbc8?}, {0x38abe40, 0x0, 0x472fb1?}, 0xc047eb4180, 0xc047eb4190)
	/home/runner/work/rollkit/rollkit/mempool/clist_mempool.go:628 +0x80f
github.com/rollkit/rollkit/state.(*BlockExecutor).commit(_, {_, _}, {{{0xb, 0x0}, {0x23349cd, 0x6}}, {0x2324885, 0x4}, 0x1, ...}, ...)
	/home/runner/work/rollkit/rollkit/state/executor.go:312 +0x61f
github.com/rollkit/rollkit/state.(*BlockExecutor).Commit(_, {_, _}, {{{0xb, 0x0}, {0x23349cd, 0x6}}, {0x2324885, 0x4}, 0x1, ...}, ...)
	/home/runner/work/rollkit/rollkit/state/executor.go:240 +0xab
github.com/rollkit/rollkit/block.(*Manager).trySyncNextBlock(0xc01bc8c5a0, {0x28e8968, 0xc01ba3a0a0}, 0x1)
	/home/runner/work/rollkit/rollkit/block/manager.go:440 +0x578
github.com/rollkit/rollkit/block.(*Manager).SyncLoop(0xc01bc8c5a0, {0x28e8968, 0xc01ba3a0a0}, 0x0?)
	/home/runner/work/rollkit/rollkit/block/manager.go:376 +0xa25
github.com/rollkit/rollkit/node.(*FullNode).OnStart.func7()
	/home/runner/work/rollkit/rollkit/node/full.go:392 +0xb8
github.com/rollkit/rollkit/types.(*ThreadManager).Go.func1()
	/home/runner/work/rollkit/rollkit/types/threadmanager.go:26 +0x87
created by github.com/rollkit/rollkit/types.(*ThreadManager).Go in goroutine 3280
	/home/runner/work/rollkit/rollkit/types/threadmanager.go:24 +0xe5
FAIL	github.com/rollkit/rollkit/node	4.015s
```

--- a/node/full_client_test.go
+++ b/node/full_client_test.go
@@ -834,6 +834,7 @@ func TestMempool2Nodes(t *testing.T) {
 	app.On("ProcessProposal", mock.Anything, mock.Anything).Return(&abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_ACCEPT}, nil)
 	app.On("CheckTx", mock.Anything, &abci.RequestCheckTx{Tx: []byte("bad")}).Return(&abci.ResponseCheckTx{Code: 1}, nil)
 	app.On("CheckTx", mock.Anything, &abci.RequestCheckTx{Tx: []byte("good")}).Return(&abci.ResponseCheckTx{Code: 0}, nil)
+	app.On("CheckTx", mock.Anything, &abci.RequestCheckTx{Tx: []byte("good"), Type: abci.CheckTxType_Recheck}).Return(&abci.ResponseCheckTx{Code: 0}, nil).Maybe()
 	key1, _, _ := crypto.GenerateEd25519Key(crand.Reader)
 	key2, _, _ := crypto.GenerateEd25519Key(crand.Reader)
 	signingKey2, _, _ := crypto.GenerateEd25519Key(crand.Reader)

